subsys/bluetooth/audio/ascs.c

```diff
@@ -339,6 +339,8 @@ static void ase_exit_state_streaming(struct bt_ascs_ase *ase)
 {
 	struct bt_bap_stream *stream = ase->ep.stream;
 	struct bt_bap_stream_ops *ops;
+	const enum bt_bap_ep_state next_state = ase->state_pending;
```

Comment:
Since the state is always updated before this is called, you could do `ascs_ep_get_state` instead.